src/app/CommandSender.cpp

```diff
@@ -116,45 +195,59 @@ void CommandSender::Close()
     }
 }
 
-CHIP_ERROR CommandSender::ProcessCommandDataIB(CommandDataIB::Parser & aCommandElement)
+CHIP_ERROR CommandSender::ProcessInvokeResponse(InvokeResponseIB::Parser & aInvokeResponse)
 {
     CHIP_ERROR err = CHIP_NO_ERROR;
-    chip::ClusterId clusterId;
-    chip::CommandId commandId;
-    chip::EndpointId endpointId;
+    ClusterId clusterId;
+    CommandId commandId;
+    EndpointId endpointId;
     // Default to success when an invoke response is received.
     StatusIB statusIB;
 
-    {
-        CommandPathIB::Parser commandPath;
-
-        err = aCommandElement.GetCommandPath(&commandPath);
-        SuccessOrExit(err);
-
-        err = commandPath.GetClusterId(&clusterId);
-        SuccessOrExit(err);
-
-        err = commandPath.GetCommandId(&commandId);
-        SuccessOrExit(err);
-
-        err = commandPath.GetEndpointId(&endpointId);
-        SuccessOrExit(err);
-    }
-
     {
         bool hasDataResponse = false;
-        chip::TLV::TLVReader commandDataReader;
+        TLV::TLVReader commandDataReader;
 
-        StatusIB::Parser statusIBParser;
-        err = aCommandElement.GetStatusIB(&statusIBParser);
+        CommandStatusIB::Parser commandStatus;
+        err = aInvokeResponse.GetStatus(&commandStatus);
         if (CHIP_NO_ERROR == err)
         {
-            err = statusIBParser.DecodeStatusIB(statusIB);
+            CommandPathIB::Parser commandPath;
+            commandStatus.GetPath(&commandPath);
+            err = commandPath.GetClusterId(&clusterId);
+            SuccessOrExit(err);
+
+            err = commandPath.GetCommandId(&commandId);
+            SuccessOrExit(err);
+
+            err = commandPath.GetEndpointId(&endpointId);
+            SuccessOrExit(err);
+
+            StatusIB::Parser status;
+            commandStatus.GetErrorStatus(&status);
+            err = status.DecodeStatusIB(statusIB);
+            SuccessOrExit(err);
         }
         else if (CHIP_END_OF_TLV == err)
         {
             hasDataResponse = true;
-            err             = aCommandElement.GetData(&commandDataReader);
+
+            CommandDataIB::Parser commandData;
+            CommandPathIB::Parser commandPath;
+            err = aInvokeResponse.GetCommand(&commandData);
+            SuccessOrExit(err);
+            commandData.GetPath(&commandPath);
+
+            err = commandPath.GetClusterId(&clusterId);
+            SuccessOrExit(err);
```

Comment:
It seems there is no extra processing in exit, we can just use ReturnErrorOnFailure instead of SuccessOrExit here

Reply:
Resolved.